### ISSUE TITLE
Can't type text on Messenger and Facebook

### ISSUE THREAD

**Issue by Schn4pper** [MrAlex94/Waterfox-Old#64]
### What happened?

Since several days I've been unable to write any messages on Messenger and any comment on Facebook. I can click on the text area to activate it (the typing mark starts blinking) but no character shows when I type. It happens on Windows and Linux on the latest version (2022.01 (64 bits)).

### Reproducible?

- [ ] I have tested this on Firefox.

### Version

G3.𝑋 (Current)

### What platform are you seeing the problem on?

Linux, Windows

**Comment by lokpro** [comment]
I meet this problem too. I can reproduce it on Firefox 56.

Facebook web is not using a < input > there. It is using a < div> to listen for key inputs to do some some tricks.
I think there is something the old browser does not support.
I have looked in the F12 for a while but have given up.

I only know I can ctrl+v to paste any character in the field first, then it can be typed in normally.

**Comment by reaper666735** [comment]
yup same here ....

**Comment by rebop** [comment]
And here as well. Very disappointing.

**Comment by condoghost** [comment]
Yes, and still an 'issue' 3-months later in Waterfox Classic. Can paste in Messenger but cannot 'type'!

**Comment by Kumba42** [comment]
You cannot type in the input field if it is empty, but as soon as there is at least one letter of input (such as from clicking "Reply" on a comment to insert someone's name), then you can type.  In this scenario, though, using the backspace or delete keys causes the page to jump to try and position the input field at the top or bottom of the screen (whichever it is closest to).  More recently, Facebook appears to have completely broken arrow key navigation in a text field in Waterfox Classic.  The cursor can no longer be moved around manually, but you must instead use the mouse to select whatever character exists before the point where you want to type -- this seems to allow the cursor bar to move, but not always.

The fastest way is to just open a notepad window and type your comment or such there, then cut & paste it into whatever input field you're trying to type in.  That, so far, appears to still work.

**Comment by timur-g** [comment]
Still an issue in WF. Very visible!

**Comment by hockeymikey** [comment]
I'm perplexed by this issue. Why does pasting work but a keypress doesn't? I tried going through the breakpoint for the keydown event in Firefox and they look nearly identical between pressing any key and a paste (keep in mind, obfuscated minified code is hard to read). There is a switch it looks like for handling paste, copy and cuts, so maybe that helps it along? Not seeing anything in APIs I've been by that suggestion something 57+ that maybe we aren't supporting. If I take too long on a breakpoint in Firefox my keypress will not be registered (think 1-2mins).
If anyone has suggestions for me to poke around and find the issue, I'm all ears!

The work around is just copy some text and paste it to get the input text box and then type in there and remove the garbage text you pasted. Alittle annoying but not the end of the world.
